src/core/unit.c

```diff
@@ -3453,6 +3453,7 @@ int unit_patch_contexts(Unit *u) {
 
                         ec->private_tmp = true;
                         ec->remove_ipc = true;
+                        ec->private_devices = true;
```

Comment:
From systemd.exec.xml , PrivateDevices=
"This setting is implied if <varname>DynamicUser=</varname> is set"

And yes after more thoughts this is a bit restrictive, however it also goes with the current concept of DynamicUser= , maybe DynamicUser= should have like no, yes, strict ?

Anyway dropped that patch for now.